jmeter-java-dsl/src/main/java/us/abstracta/jmeter/javadsl/core/controllers/DslWeightedSwitchController.java

```diff
@@ -0,0 +1,107 @@
+package us.abstracta.jmeter.javadsl.core.controllers;
+
+import com.blazemeter.jmeter.control.WeightedSwitchController;
+import com.blazemeter.jmeter.control.WeightedSwitchControllerGui;
+import org.apache.jmeter.gui.util.PowerTableModel;
+import org.apache.jmeter.testelement.TestElement;
+import org.apache.jorphan.collections.HashTree;
+import us.abstracta.jmeter.javadsl.core.BuildTreeContext;
+import us.abstracta.jmeter.javadsl.core.testelements.BaseTestElement;
+import us.abstracta.jmeter.javadsl.core.threadgroups.BaseThreadGroup.ThreadGroupChild;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * DSL wrapper for {@link WeightedSwitchController}
+ */
+public class DslWeightedSwitchController extends BaseController {
+
+  private final PowerTableModel model;
+  private final WeightedSwitchController wsc;
+
+  public DslWeightedSwitchController() {
+    super("Weighted Switch Controller",
+        WeightedSwitchControllerGui.class,
+        new ArrayList<>()
+    );
+
+    model = new PowerTableModel(
+        new String[]{"Name", "Weight", "Enabled"},
+        new Class[]{String.class, String.class, String.class}
+    );
+
+    wsc = new WeightedSwitchController();
+  }
+
+  /**
```

Comment:
no need to document private methods if you don't want to, they are not visible to API users.